impl/src/main/java/com/stormpath/sdk/impl/account/DefaultAccount.java

```
@@ -223,6 +222,74 @@ public GroupMembership addGroup(Group group) {
     }
 
     @Override
+    public GroupMembership addGroup(String hrefOrName) {
+        Group group =  findGroup(hrefOrName);
+        if (group != null){
+            return DefaultGroupMembership.create(this, group, getDataStore());
+        }
+        return null;
+    }
+
+    private Group findGroup(String hrefOrName) {
+
+        Group group = null;
+
+        //Let's check if hrefOrName looks like an href
+        String[] splitHrefOrName = hrefOrName.split("/");
+        if (splitHrefOrName.length > 4) {
+            try {
+                group = getDataStore().getResource(hrefOrName, Group.class);
+            } catch (ResourceException e) {
+                // Although hrefOrName seemed to be an actual href value no Resource was found in the backend.
+                // Maybe this is actually a name rather than an href
+            }
+        }
+
+        // Notice that accounts can only be added to Groups in the same directory
+        Directory directory = this.getDirectory();
+        if (group != null && group.getDirectory().getHref().equalsIgnoreCase(directory.getHref())){
+            return group;
+        }
+
+        GroupList groups = directory.getGroups(Groups.where(Groups.name().eqIgnoreCase(hrefOrName)));
+        if (groups.iterator().hasNext()){
+            group = groups.iterator().next();
+        }
+
+        return group;
+    }
+
+    @Override
+    public Account removeGroup(Group group) {
+        GroupMembership groupMembership = null;
+        for (GroupMembership aGroupMembership : getGroupMemberships()) {
+            if (aGroupMembership.getGroup().getHref().equalsIgnoreCase(group.getHref())) {
+                groupMembership = aGroupMembership;
+                break;
+            }
+        }
+        if (groupMembership != null){
+            groupMembership.delete();
+        }
```

I think this method should throw an exception if the account has not been removed from the account (the account was not member of this group) in order to bring this fact to the developer's attention. Otherwise he could be wrongly assuming that he has successfully removed the account from a group when in fact he has not removed it.